proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,471 @@
+1. It meets the criteria specified in the contribution guide. See the
+   [appendix](#preliminary-review-criteria) for details.
+
+If the criteria are not satisfied, the maintainers will comment on the issue
+specifically indicating what work is still required. If the contributor has
+questions, they should refer to the contributing guide documentation or ask the
+maintainers to clarify. Once the contributor makes the required changes, the
+maintainers will follow up within 5 business days.
+
+If the contributor is no longer actively engaged by responding to comments from
+maintainers or making code changes for more than 30 business days, the
+contribution is stale. If not, the pull request will be closed and the
+corresponding issue will be marked as ready for contribution.
+
+### Ready for review stage
+
+Once the preliminary review criteria from the implementation stage have been
+met, the tracking GitHub issue is labeled as ready for review and added to a
+queue for maintainer review. The maintainers will select issues for full review
+from this queue. The issues will be prioritized for review publicly based on the
+following criteria in descending order of importance:
+
+1. Bugs, documentation changes, and feature requests will be prioritized in that
+   order.
+1. The time available to the maintainers will be considered against the
+   complexity of the contribution.
+1. Older contributions will be selected before more recent ones.
+
+A maintainer must select the issue from the queue with the current highest
+priority. The prioritization of issues in the queue is dynamic, reviewed on a
+regular cadence, and is ultimately decided based on the maintainers' discretion.
+Having a pull request selected for review does not guarantee how long it will be
+until it is merged.
+
+### Review stage
+
+Selecting a pull request for review will move the tracking GitHub issue to the
+under review stage. The maintainers will limit the number of active reviews to
+[TBD]. This criteria for accepting a pull request includes:
+
+1. The change is made in the right place and fits in the existing architecture.
+1. The change does not increase the maintainability in an unreasonable fashion.
+1. There is sufficient documentation, both in the code and for end users.
+1. The code is readable and follows best practices and conventions.
+
+If a pull request is not ready to merge, the maintainers will request feedback
+and changes from the contributor by asking specific questions or specifying
+criteria that should be met. If changes are requested, the contributor should
+make the changes within [TBD] business days. If the contributor does not respond
+within [TBD] days, the pull request will be marked as stale and the maintainers
+will complete the pull request.
+
+Once a pull request is completed to the satisfaction of the maintainers, it will
+be approved and merged. The contribution will be labeled to acknowledge it as a
```

Comment:
We should also note that we close the corresponding tracking GitHub issue once the PR is merged.